### ISSUE TITLE
Hosting der App unter Subdomain

### ISSUE THREAD

**Issue by slamdown251** [solectrus/hosting#11]
Hallo Herr Ledermann,

wirklich ein tolles Projekt!
Ich habe eine Frage: Ist es möglich die Mainapp unter einer Subdomain zu hosten? Weil aktuell zeigen die Links des Buttons Verlauf und des Solectrus-Logos auf die Root-Domain meines Servers. Ich habe schon versucht den APP_HOST in der .env-Datei anzupassen, das hat aber leider keinen Erfolg gebracht.

**Comment by ledermann** [comment]
@slamdown251 Ja, das sollte klappen, wobei der Reverse Proxy entscheidend ist. Sofern der richtig konfiguriert ist, läuft Solectrus auch auf einer Subdomain, wie das auch beispielsweise bei meiner eigenen Instanz der Fall ist: https://solectrus.ledermann.dev (ich verwende Traefik als Reverse Proxy)

Die Umgebungsvariable `APP_HOST` sollte dabei auf die volle Domain gesetzt werden, also z.B. `solectrus.example.org`, wobei das aber nur für Spezialfälle (wie einer CDN-Nutzung) wirklich erforderlich ist.

Ich habe jetzt gerade noch eine kleine Korrektur vorgenommen, sodass die Links zum Verlauf (und Top10) nun relativ sind, also nicht mehr den Host enthalten. Stutzig macht mich jedoch der Hinweis, dass auch der Klick auf das Logo zur root-Domain führt - denn der ist schon immer relativ gewesen.

Zur weiteren Fehleranalyse empfehle ich, auf das `develop`-image zu wechseln, also im `docker-compose.yaml` statt `ghcr.io/solectrus:latest` jetzt `ghcr.io/solectrus/solectrus:develop` zu verwenden. Möglicherweise sind damit einige Probleme behoben, vielleicht aber noch nicht alle.

Gerne sehe ich mir auch Ihre Instanz genauer an, wenn Sie mir die Domain verraten möchten.

**Comment by slamdown251** [comment]
Danke für die schnelle Antwort, hier liegt der Fehler bei mir, ich habe mich falsch ausgedrückt, ich möchte Solectrus nicht unter einer Subdomain hosten sondern quasi als Subfolder, also im Stil `https://xyz.net/solectrus`. Wenn ich den `APP_HOST` dementsprechend anpasse also z.B. auf `xyz.net/solectrus` hat das aber keine Auswirkung. Ist es möglich dem internen Linkbuilder mitzuteilen, dass er bei der Generierung der Links die Subfolder mit berücksichtigt?

**Comment by ledermann** [comment]
Denkbar wäre das, aber derzeitig unterstützt Solectrus das nicht. Mit der aktuellen Version gibt es nur die Möglichkeiten Root-Domain oder Sub-Domain. Ich würde sehr empfehlen, eine Subdomain zu verwenden.

Ich lasse das Issue mal offen, ggfs. ergänze ich dieses Feature später noch.

**Comment by slamdown251** [comment]
Alles klar, es würde mich sehr freuen wenn das Feature eines Tages kommen würde, trotzdem vielen Dank für Ihre bisherige Arbeit!
